sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AnsiTypeCoercion.scala

```diff
@@ -173,6 +185,8 @@ object AnsiTypeCoercion extends TypeCoercionBase {
       inType: DataType,
       expectedType: AbstractDataType): Option[DataType] = {
     (inType, expectedType) match {
+      case (_: StringType, st: StringType) =>
+        Some(st)
```

Comment:
Can you use explicit `isDefaultCollation`? In general, let's stay away of using `StringType` case object. We left it due to backwards compatability, but we should use explicit `st: StringType if st.isDefaultCollation` checks in Spark code.

Reply:
I am not sure I get this. We need to add the rule for casting any StringType class as acceptsType below would just lend us into keeping original collationId, and we would not be able to cast different collations into one.

Reply:
I misread the code. My bad. Everything looks good.